phoenix-hbase-compat-2.0.1/pom.xml

```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project
+  xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation=
+    "http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+    <parent>
+    <groupId>org.apache</groupId>
+    <artifactId>apache</artifactId>
+    <version>21</version>
+  </parent>
+  <groupId>org.apache.phoenix</groupId>
+  <artifactId>phoenix-hbase-compat-2.0.1</artifactId>
+  <version>5.1.0-SNAPSHOT</version>
+  <name>Phoenix Hbase 2.0.1 compatibility</name>
+  <description>Compatibility module for HBase 2.0.1 - 2.0.4 and 2.1.0 - 2.1.1</description>
+
+  <build>
+    <plugins>
+      <!-- Setup eclipse -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-eclipse-plugin</artifactId>
+        <version>2.9</version>
+        <configuration>
+          <buildcommands>
+            <buildcommand>org.eclipse.jdt.core.javabuilder</buildcommand>
+          </buildcommands>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.rat</groupId>
+        <artifactId>apache-rat-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+
+  <dependencies>
+       <!-- HBase dependencies -->
+      <dependency>
+        <groupId>org.apache.hbase</groupId>
+        <artifactId>hbase-server</artifactId>
+        <version>2.0.1</version>
```

Comment:
Should be `hbase.compat.version`?

Reply:
Thinking through this some more, I think this goes to my `hbase.profile` suggestion in the other thread.

`hbase.profile` activates the 2.y release "compat line" we want, and pull in that code. `hbase.version` sets the specific version of hbase we depend on. We'll automatically set one for the compat module which can be the latest in that 2.y.z release line (we should just use the latest .z -- often times hbase will fix things which we would see breaking in our tests).

Is that what you see? Stating it out lout to make sure we're on the same page :)

Reply:
> Should be `hbase.compat.version`?

My intent is to always build (and make available in the maven repo) each of the compatibility modules, so that one can use the version-independent unshaded JARs/artifacts plus the right compatibility module, to get a working phoenix classpath. (even though it is not really a use case that we expect in the real world)

specifying hbase.compat.version here would break building the compatibility module for the non-selected versions.

Reply:
Right you are. Thanks for catching that :)